pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -524,7 +534,7 @@ def add_qiskit_data(
                 q_ctrl_box = _get_qcontrol_box(c_gate=instr, params=params)
                 self.tkc.add_qcontrolbox(q_ctrl_box, qubits)
 
-            elif isinstance(instr, (Initialize, StatePreparation)):
+            elif optype == OpType.StatePreparationBox:
```

Comment:
Seems more correct to match the optype as both of the qiskit instructions can be converted to a `StatePreparationBox` instance.

Reply:
In fact we could probably rewrite `CircuitBuilder.add_qiskit_data` to just check the `optype` and not look at the qiskit `Instruction`. Its just the handling of `UnitaryGate` and `PauliEvolutionGate` that are preventing this at the moment.